api/src/main/java/javax/config/ConfigValue.java

```diff
@@ -192,8 +209,8 @@
     String getKey();
 
     /**
-     * Returns the actual key which led to successful resolution and corresponds to the resolved value. This applies
-     * only when {@link #withLookupChain(String...)} is used.
+     * Returns the actual key which led to successful resolution and corresponds to the resolved value.
+     * This is mostly useful when {@link #withLookupChain(String...)} or {@link #evaluateVariables(boolean)} is used.
```

Comment:
What if this key is based on multiple variables? Which one is the "resolved one"?

Reply:
The last effective String.
So if you have cfg.access("myconfigkey").withLookupChain("${app.projectstage}") 
and app.projectstage resolves to 'Prodution' you might end up with 'myconfigkey.Production'